api/lib/infrastructure/repositories/campaign-profiles-collection-participation-summary-repository.js

```diff
@@ -79,4 +80,11 @@ async function _makeMemoizedGetPlacementProfileForUser(results) {
   return (userId) => placementProfiles.find((placementProfile) => placementProfile.userId === userId);
 }
 
+function _filterQuery(qb, filters) {
+  if (filters.divisions) {
+    const divisionsLowerCase = filters.divisions.map((division) => `'${division.toLowerCase()}'`);
+    qb.whereRaw(`LOWER("schooling-registrations"."division") in (${divisionsLowerCase.join(',')})`);
```

Comment:
En lisant l'algo, j'ai l'impression qu'il y a un pb avec les quotes dans la génération de la requête :
```
const divisionsLowerCase = filters.divisions.map((division) => `'${division.toLowerCase()}'`);
```
Si on a `divisions = ["A", "B"]`, il sera mappé en `divisionsLowerCase=["'a'", "'b'"]`
Puis le `join` dans le where
```
qb.whereRaw(`LOWER("schooling-registrations"."division") in (${divisionsLowerCase.join(',')})`);
```
On obtiendra:
```
LOWER("schooling-registrations"."division") in (''a'',''b'') // 2 fois les double quote
```

Reply:
Bien vu !! Je corrige c'est assez ouf que ça marche quand même !

Reply:
Je viens de tester
```js
//Cette version marche pas
const divisionsLowerCase = filters.divisions.map((division) => { division.toLowerCase() } );
console.log(`LOWER("schooling-registrations"."division") in (${divisionsLowerCase.join(',')})`);
//LOWER("schooling-registrations"."division") in (barry,white)

//==================================================================
const divisionsLowerCase = filters.divisions.map((division) => `'${division.toLowerCase()}'`);
console.log(`LOWER("schooling-registrations"."division") in (${divisionsLowerCase.join(',')})`);
//LOWER("schooling-registrations"."division") in ('barry','white')

```

Reply:
Le **`** n'ajoute pas les ' dans la de caractères donc on doit les rajouter, par contre c'est clairement pas évident si tu as une autre idée d'implémentation je suis preneur

Reply:
Est-ce qu'on ne peut pas utiliser [`whereIn`](http://knexjs.org/#Builder-whereIn) de `knex` ?
```js
const divisionsLowerCase = filters.divisions.map((division) => `'${division.toLowerCase()}'`);
qb.whereIn('LOWER("schooling-registrations"."division")', divisionsLowerCase);
```

Reply:
👋 Je vois passer ce `whereRaw` et ça me donne envie de rappeler qu'on se débrouille normalement pour ne jamais concaténer de données dans le SQL. On a même une règle ESLint pour ça  https://github.com/1024pix/pix/blob/68f7b3b59b8017292b1d7504af042638304513a9/api/.eslintrc.yaml#L10 mais malheureusement elle ne détecte pas l'utilisation de `whereRaw`…

Dans les cas (relativement rares) où on se retrouve à utiliser `knex.raw` ou équivalent, on continue d'utiliser le mécanisme de passage de paramètres par _bindings_ de `knex`.

Par exemple avec ``qb.whereRaw(`LOWER("schooling-registrations"."division") IN (${divisionsLowerCase.map('?').join(',')})`, divisionsLowerCase);``. On n'injecte plus de données dans la chaîne. Mais ça reste un peu suspect (difficile à auditer) parce qu'il y a quand même une chaîne SQL dynamique.

Sinon, la proposition de @bpetetot est presque bonne, mais par défaut `whereIn` va considérer son premier argument comme un identifiant de colonne ce que `LOWER("schooling-registrations"."division")` n'est pas. En revanche si on l'encadre dans `knex.raw` ça devrait passer : ``qb.whereIn(knex.raw('LOWER("schooling-registrations"."division")'), divisionsLowerCase)``. Là encore on évite de mettre les données dans du `raw`.

On peut aussi utiliser la possibilité de passer tout le tableau sous la forme d'un unique paramètre : ``qb.whereRaw(`LOWER("schooling-registrations"."division") = ANY(:divisionsLowerCase)`, { divisionsLowerCase })``. Cette dernière option est vraisemblablement plus efficace que les deux précédentes parce que le SQL généré ne contient pas une longue liste de paramètres (`IN (?,?,?,?,…)`) qui rend la requête plus volumineuse et différente à chaque appel ce qui est gênant pour le cache.

Reply:
J'ai pris l'implémentation :
```js
qb.whereRaw(`LOWER("schooling-registrations"."division") = ANY(:divisionsLowerCase)`, { divisionsLowerCase })
```
Et j'en ai profité pour corrigé l'implémentation côté `pix/api/lib/infrastructure/repositories/campaign-assessment-participation-summary-repository.js:127`